test/test_np_chunker.py

```diff
@@ -0,0 +1,135 @@
+            ('chairman', 'NN', 'B'), ('of', 'IN', 'O'),
+            ('Consolidated', 'NNP', 'B'), ('Gold', 'NNP', 'I'),
+            ('Fields', 'NNP', 'I'), ('PLC', 'NNP', 'I'), (',', ',', 'O'),
+            ('was', 'VBD', 'O'), ('named', 'VBN', 'O'), ('a', 'DT', 'B'),
+            ('nonexecutive', 'JJ', 'I'), ('director', 'NN', 'I'),
+            ('of', 'IN', 'O'), ('this', 'DT', 'B'), ('British', 'JJ', 'I'),
+            ('industrial', 'JJ', 'I'), ('conglomerate', 'NN', 'I'),
+            ('.', '.', 'O')]]
+
+    def test_chunk_sents_iob(self):
+        '''
+        The chunker returns reasonable IOB labels
+        '''
+        text_tags = [[(t[0], t[1]) for t in sent]
+            for sent in self.text_tags_iob]
+        iob_labels = chunker.chunk_sents(text_tags, True)
+        self.assertListEqual(self.text_tags_iob, iob_labels)
+
+    def test_chunk_sents(self):
+        '''
+        The chunker returns noun phrases
+        '''
+        text_tags = [[(t[0], t[1]) for t in sent] 
+            for sent in self.text_tags_iob]
+        noun_phrases = chunker.chunk_sents(text_tags)
+        expected_phrases = [
+    [[('Pierre', 'NNP'), ('Vinken', 'NNP')],
+    [('61', 'CD'), ('years', 'NNS')],
+    [('the', 'DT'), ('board', 'NN')],
+    [('a', 'DT'), ('nonexecutive', 'JJ'), ('director', 'NN')],
+    [('Nov.', 'NNP'), ('29', 'CD')]],
+
+    [[('Mr.', 'NNP'), ('Vinken', 'NNP')],
+    [('chairman', 'NN')],
+    [('Elsevier', 'NNP'), ('N.V.', 'NNP')],
+    [('the', 'DT'), ('Dutch', 'NNP'), ('publishing', 'NN'), ('group', 'NN')]],
+
+    [[('Rudolph', 'NNP'), ('Agnew', 'NNP')],
+    [('55', 'CD'), ('years', 'NNS')],
+    [('chairman', 'NN')],
+    [('Consolidated', 'NNP'),
+    ('Gold', 'NNP'),
+    ('Fields', 'NNP'),
+    ('PLC', 'NNP')],
+    [('a', 'DT'), ('nonexecutive', 'JJ'), ('director', 'NN')],
+    [('this', 'DT'),
+    ('British', 'JJ'),
+    ('industrial', 'JJ'),
+    ('conglomerate', 'NN')]]]
+
+        self.assertEqual(noun_phrases, expected_phrases)
+    
+    def test_empty(self):
+        '''
+        The chunker works OK if it's passed empty tokens, or empty sentences
+        '''
+        sentences = [
+            [],
+            [('The', 'DT'), ('empty', 'NN'), ('', 'NN'), ('POS', ''), ('', ''),
+                ('.', '.')]]
+        chunks = chunker.chunk_sents(sentences, True)
+        self.assertEqual(
+            chunks,
+            [[], 
+            [('The', 'DT', 'B'),
+              ('empty', 'NN', 'I'),
+              ('', 'NN', 'I'),
+              ('POS', '', 'O'),
+              ('', '', 'O'),
+              ('.', '.', 'O')]]
+        )
+
+        np_chunks = chunker.chunk_sents(sentences)
+        self.assertEqual(
+            np_chunks,
+            [[], [[('The', 'DT'), ('empty', 'NN'), ('', 'NN')]]])
+
+    def test_unicode(self):
+        sentence = u'Beyonc\u00e9 performed during half time of Super Bowl'
+        sentence += ' XLVII .'
```

Comment:
It's odd that you're adding a non-unicode string to the end of a unicode string. I realize this happens to work, but I would probably make the second string a `u''` just for clarity. Also, do you intend to have a space between "XLVII" and the "."? That seems weird to me. Also, note that Python allows implicit string concatenation with just spaces for string literals:
```Python
>>> s = (u'The ' u'beginning')
>>> s
u'The beginning'
```
This means you could also break the line this way:
```Python
sentence = (u'Beyonc\u00e9 performed during half time of Super Bowl'
    u' XLVII .')
```
or even this way:
```Python
sentence = (
    u'Beyonc\u00e9 performed during half time of Super Bowl'
    u' XLVII .'
)
```